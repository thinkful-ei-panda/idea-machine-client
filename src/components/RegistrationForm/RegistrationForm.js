import React from 'react'
import config from '../../config'
import TokenService from '../../services/token-service'

class RegistrationForm extends React.Component {
  state = {
    error:null
  }

  handleRegistrationSubmit = (ev) => {
    ev.preventDefault()
    this.setState({error:null})
    const {username, password} = ev.target
    fetch(`${config.API_ENDPOINT}/users`, {
      method: 'POST',
      headers: {
        'content-type' : 'application/json'
      },
      body: JSON.stringify({
        user_name:username.value,
        password:password.value
      })
    })
    .then(res => (!res.ok)
    ? res.json().then(e => Promise.reject(e))
    : res.json())
    .then(() => {

      //After registering, logs in
      fetch(`${config.API_ENDPOINT}/auth/login`, {
        method: 'POST',
        headers: {
          'content-type' : 'application/json'
        },
        body: JSON.stringify({
          user_name:username.value,
          password:password.value
        })
      })
      .then(res => (!res.ok)
      ? res.json().then(e => Promise.reject(e))
      : res.json())
      .then(res => {
        TokenService.saveAuthToken(res.authToken)
        username.value = ''
        password.value = ''

        //toggle logged in state
        this.props.handleLogIn(this.props.history)
      })
      .catch(error => {
        this.setState({error})
      })      
    })
    .catch(error => {
      this.setState({error})
    })
  }


  render() {
    const {error} = this.state

    return (
      <div className="login-container">
        <form onSubmit={this.handleRegistrationSubmit} className="login-form">
          <legend></legend>
          <fieldset>
            <div className="column">
              <div>
                <label htmlFor="username">Username:</label>
                <input name="username" id="username" />
              </div>
              <div>
                <label htmlFor="password">Password:</label>
                <input name="password" id="password" />
              </div>
              <div>
                <button>Register</button>
              </div>
            </div>
          </fieldset>
        </form>
        {error && <div>{error.error}</div>}
      </div>
    )
  }  
}

export default RegistrationForm