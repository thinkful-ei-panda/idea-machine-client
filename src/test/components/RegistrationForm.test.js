import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import RegistrationForm from '../../components/RegistrationForm/RegistrationForm';
describe(`RegistrationForm Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <RegistrationForm />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <RegistrationForm />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})