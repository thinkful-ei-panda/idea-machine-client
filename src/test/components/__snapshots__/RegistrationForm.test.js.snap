// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="login-container"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <legend />
    <fieldset>
      <div
        className="column"
      >
        <div
          className="formInputContainer"
        >
          <label
            htmlFor="username"
          >
            Username
          </label>
          <input
            id="username"
            name="username"
          />
        </div>
        <div
          className="formInputContainer"
        >
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            id="password"
            name="password"
          />
        </div>
        <div
          className="buttonContainer"
        >
          <button>
            Register
          </button>
        </div>
      </div>
    </fieldset>
  </form>
</div>
`;
